### PR TITLE
Silly tests

### PR DESCRIPTION
**Pull request by mjgpy3**
Probably the silliest possible approach. Should be easy to evolve into something crazier if we want to.

:clown_face:

Here's an example, false negative, failure
``` lisp
.......
Expected
  (if (lambda (x) x) 1)
to be
  : 2
but it was
  (if (lambda (x) x) 1) : #<unspecified>

```

### CODE REVIEW

File: dytype/dytype.scm [hunk 1]
```diff
@@ -141,4 +145,6 @@
   (display "dytype >>\n")
   )
 
-(repl)
+(if (null? (command-line-arguments))
+  (repl)
+  (run-and-print-with prelude (call-with-input-string (car (command-line-arguments)) read)))
```
Comment (mjgpy3) [inline dytype/dytype.scm:150]: If the exe's given an arg it'll evaluate it. If not, spin off into the repl.

File: dytype/test_dytype.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+assert_n 1 '(if 1 1)'
+assert_unspecified '(if (lambda (x) x) 1)'
+assert_n 1 '(if #t 1 1)'
+assert_n 2 '(if #t (lambda (x) 42) (lambda (y) 99))'
+assert_unspecified '(if #t (lambda (x) 42) 99)'
+assert_unspecified '(if 1 2 3 4)'
+assert_unspecified '(if 1 (lambda (x)))'
+
+assert_n 1 "`cat examples/fizzbuzz.dy`"
```
Comment (mjgpy3) [inline dytype/test_dytype.sh:50]: Hehe :man_office_worker: :briefcase: :pen:  :white_large_square:

File: dytype/test_dytype.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+assert_unspecified() {
+  if grep --quiet ': #<unspecified>' <(./dytype "$@"); then
```
Comment (mjgpy3) [inline dytype/test_dytype.sh:4]: This and `assert_n` should probably use regex `$` to ensure it ends with the correct type.

File: dytype/test_dytype.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    printf '.'
+  else
+    echo
+    echo "Expected"
+    echo "  $@"
+    echo "to be"
+    echo "  : unspecified"
+    echo "but it was"
+    echo "  $(./dytype "$@")"
```
Comment (Kazark) [inline dytype/test_dytype.sh:13]: Why re-run? Don't we just want to save it off the result from the first run in a variable?

**Review by Kazark — COMMENTED** [review]
Shell-based automated testing! I love it! :joy_cat:

**Review by Kazark — APPROVED** [review]

**Comment by Kazark** [comment]
Next step: plug in some CI.

**Pull request merged**